### ISSUE TITLE
Steam controller not working with Steam big picture games on streaming or on desktop

### ISSUE THREAD

**Issue by teletrail** [ValveSoftware/steam-for-linux#5217]
#### Your system information

* Steam client version (build number or date): oct 19, 2017
* Distribution (e.g. Ubuntu): Dbian Sid/Buster
* Opted into Steam client beta?: [Yes/No] yes
* Have you checked for system updates?: [Yes/No] yes

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)

when I have the controller plugged in (or connected via wireless) I can navigate big picture mode, but input is non functional in games. tried with cryptark and assault android cactus, both games claiming full compatibility with steam controller.

any support would be appreciated, thanks.

#### Steps for reproducing this issue:

1. plug in controller
2. run big picture, run games
3. controller doesn't function

**Comment by hamishmaku** [comment]
@kisak-valve it also doesn't work in-game with Portal 2, so it's not only with 3rd party games. problem persists.

**Comment by kisak-valve** [comment]
Hello @hamishmaku, did you install steam with the distro-provided steam package or the valve-provided steam package? If you're using the distro-provided steam package, do you also have the steam-devices system package installed?

**Comment by hamishmaku** [comment]
distro-provided, but i have steam-devices installed. controller works in
big picture mode but not in games. mouse emulation functions if enabled.

thanks

**Comment by Jusix** [comment]
Same here, did just spend 47€ for a disfuntional controller. I can only navigate through big picture mode, but neither DIRT Rally, F1 2017, GRID Autosport or Rocket League works. The Controller does nothing while ingame.

**Comment by hamishmaku** [comment]
@kisak-valve problem is still present - are any solutions suggested? controller works fine in windows.

**Comment by armahillo** [comment]
Chiming in here -- I just got my Steam Link + Steam Controller today and this bug also affects me.

I have tested with:

 * Kingdom: New Lands (mixed)
 * Ziggurat (mostly non-functional)
 * Dreamfall: Chapter (works when Key/Mouse binded)

Binding buttons as Keyboard/Mouse buttons works. 

Ziggurat (FPS game) is the most puzzling. The Gyro works GREAT -- it's binded to mouselooking and is very responsive. The right trackpad (also assigned to mouse) also works fine. When I assign two buttons to be Esc/Enter, those also work. But the left joystick input does not work as joystick input. None of the other buttons work (including triggers, left pad, etc). Strangely, the menu icons SHOW the controller icons -- tapping a button on the keyboard switches all the icons automatically to keyboard shortcuts, and then tapping a controller button (even a nonfunctional one) reverts them back to the controller icons. The game clearly *sees* the controller, but the input is not being relayed correctly.

Kingdom: New Lands and Dreamfall: Chapters were both semi-successful, but only because I binded keyboard strokes to the different controls -- as a controller it does not work.

Mark of the Ninja BRIEFLY worked and appeared to intuit the button controls, but then a weird button combination briefly brought up the on-screen dual keyboard, I saw a mouse cursor, and then nothing worked at all, with the controller.

I have tried both connected to the SteamLink via Bluetooth as well as hardwired to my desktop, I do have `steam-devices` installed and rebooted after installing it. When I open Steam with the controller plugged in, it informs me "If you want to get the most out of your Steam Controller, be sure to use Big Picture Mode!"

I have also tried with both dual and single screens. I was not using OBS or any screen capturing.

System info:
* Ubuntu 17.10
* NVidia GTX 960


It'd be really great if this could be fixed. I really, really like the controller and the Steamlink, and it's very frustrating that it's not working correctly. :(

**Comment by armahillo** [comment]
Additional feedback:

I've had SUCCESS getting the Steam Controller to work, as intended, with SproggiWood. It does NOT work by default, but if I go into the settings menu, choose "Controller Input" and select "Off", then select "On" again, it works. I need to do this every time I start the game, but once I do, it works great, and the bindings shown in-game (UI hints) reflect the actual button used for that action (so "[A]" onscreen corresponds to [A] on the controller, etc).

I will see if any of my other games have declarative "I'm using a controller input!" settings, and I will try those out to see if it's similar experience. Ziggurat works tightly with Controllers but does not have an explicit "I'M USING A CONTROLLER, FAM!" setting switch.

**TL;DR: Explicitly enabling controller input in a game (Sproggiwood) results in correct behavior of the Steam Controller, but it needs to be enabled every time the game is run.**

**Comment by dacvetux** [comment]
Hi - Same here. In The game Hollow Night (for example) I don+t have such an option -> the Controller is just not detected. 

Funny thing is -> It all worked flawlessly before. Now for some months, I can use it to get to the game but not to play the game?

**Comment by 123Kai123** [comment]
Maybe I'm a little late, but I think the easiest solution to this is just adding the user to the "input"-group with this command:
`sudo usermod -a -G input [your username]`
After that you might have to log out and back in, but everything should be working then.

**Comment by kisak-valve** [comment]
Hello, is anyone still experiencing this issue on an up to date system? In general, the regular user should have access to `/dev/uinput` in addition to the udev rules provided by [steam-devices](https://github.com/ValveSoftware/steam-devices).
